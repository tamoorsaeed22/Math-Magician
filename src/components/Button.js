import React from 'react';

// eslint-disable-next-line react/prop-types
export default function Button({ value, span, bg }) {
  return (
    <button type="button" className={`btn ${[span, bg].join('')}`}>{value}</button>
  );
}
