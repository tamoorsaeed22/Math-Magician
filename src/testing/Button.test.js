import renderer from 'react-test-renderer';
import Button from '../components/Button';

function handleClick() {
  return 'message';
}

it('renders correctly', () => {
  const tree = renderer.create(
    <Button value="1" span="span-two" bg="bg-color" press={handleClick} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
