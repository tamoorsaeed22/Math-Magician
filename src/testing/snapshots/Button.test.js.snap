// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn span-twobg-color"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;